### ISSUE TITLE
Provide systemd unit file in deb and rpm packages

### ISSUE THREAD

**Issue by pavel-odintsov** [FerretDB/FerretDB#2287]
### What should be done?

Hello!

I've started playing with FerretDB and noticed lack of systemd unit in official deb:
```dpkg -L ferretdb
/usr
/usr/bin
/usr/bin/ferretdb
```

I have created my own and it works just fine:
```
[Unit]
Description=FerretDB database
After=network.target remote-fs.target
 
[Service]
Type=simple
ExecStart=/usr/bin/ferretdb --debug-addr="127.0.0.1:8089"
Restart=on-failure
RestartSec=5s

[Install]
WantedBy=multi-user.target
```

Would be great to have it in official package.

I understand that many people use Docker these days but we mostly operate bare metal / VM based setups without any containers in place.

**Comment by AlekSi** [comment]
It should be possible to add it using fnpm: https://nfpm.goreleaser.com/tips/

**Comment by pavel-odintsov** [comment]
I think this guide has great point that non root user definitely needed to run FerretDB as: https://blog.ferretdb.io/install-ferretdb-mongodb-alternative-ubuntu/

```
[Unit]
Description=FerretDB database
After=network.target remote-fs.target
 
[Service]
Type=simple
User=ferretdb
ExecStart=/usr/bin/ferretdb --debug-addr="127.0.0.1:8089"
Restart=on-failure
RestartSec=5s

[Install]
WantedBy=multi-user.target

```

One of the issues I see with shipping Systemd Unit file from RPM or DEB package is fact that type of backend is specified via command line arguments and both SQLite and PostgreSQL backends need to be supported in same time.

It may be solved via configuration file stored in some well known location in /etc and then if we can drop all arguments and keep only path to configuration file then systemd unit file for FerretDB will support all backends

**Comment by chriscz** [comment]
Below is the ansible task I use to do the setup of ferretdb.

I made use of SystemD's environment facility to pass down the password and state directory to the ferretdb process.

<details>
<summary>
<b>
Click here to see the Ansible task file
</b>
</summary>

```yaml
---
# Upload the ferretdb.deb file to the remote machine
- name: Upload deb package file
  ansible.builtin.copy:
    src: "ferretdb-linux-amd64-v1.21.0.deb"
    dest: "/tmp/ferretdb.deb"

# Installs a ferretdb.deb file from the uploaded location
- name: Install deb package file
  ansible.builtin.apt:
    deb: "/tmp/ferretdb.deb"
    state: present

# Alternative approach, if you prefer the static binary should the above deb file not work for you.
# - name: Upload ferretdb executable
#  ansible.builtin.copy:
#    src: "ferretdb-linux-amd64"
#    dest: "/usr/bin/ferretdb"
#    owner: root
#    group: root
#    mode: "0755"

# Create a passwordless, homeless service user account for ferretdb
- name: Create ferretdb user
  ansible.builtin.user:
    name: ferretdb
    system: yes
    shell: "/usr/sbin/nologin"
    home: "/nonexistent"
    create_home: no

# Create an environment file only readable by root
- name: Create /etc/ferretdb/environment
  ansible.builtin.copy:
    dest: "/etc/ferretdb/environment"
    content: |
      # FerretDB environment variables
      FERRETDB_POSTGRESQL_URL=postgres://username:password@127.0.0.1:5432
      FERRETDB_STATE_DIR=/var/lib/ferretdb
    owner: root
    group: root
    mode: "0600"

- name: Create /var/lib/ferretdb state directory
  ansible.builtin.file:
    path: "/var/lib/ferretdb"
    state: directory
    owner: ferretdb
    group: ferretdb
    mode: "0755"

# Create a systemd service file that loads the aforementioned environment file as part of booting up the ferret db process.
- name: Create systemd service file
  ansible.builtin.copy:
    dest: "/etc/systemd/system/ferretdb.service"
    content: |
      [Unit]
      Description=FerretDB service
      After=network-online.target
      Wants=network-online.target

      [Service]
      User=ferretdb
      EnvironmentFile=/etc/ferretdb/environment
      ExecStart=/usr/bin/ferretdb
      Restart=always

      [Install]
      WantedBy=multi-user.target
    owner: root
    group: root
    mode: "0644"

- name: Enable systemd service
  ansible.builtin.systemd:
    name: ferretdb
    enabled: yes

- name: Reload systemd daemon
  ansible.builtin.systemd:
    daemon_reload: yes
```
</details>

**Comment by AlekSi** [comment]
We just released v1.23.0 with experimental systemd unit files. Please let us know what you think.
